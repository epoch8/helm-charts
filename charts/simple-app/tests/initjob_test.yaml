suite: simple-app init job test

values:
  - initjob-values.yaml

release:
  name: simple
  namespace: default

chart:
  version: 0.0.0

tests:
  - it: All manifests should match snapshot
    templates:
      - initjob.yaml
    asserts:
      - matchSnapshot: {}

  - it: Init Job - Set default init job image
    templates:
      - initjob.yaml
    asserts:
      - isKind:
          of: Job
      - equal:
          path: spec.template.spec.containers[0].image
          value: test:1.1.1

  - it: Init Job - Set init job image
    templates:
      - initjob.yaml
    set:
      initJob:
        repository: "test-hook"
        tag: "2.2.2"
    asserts:
      - isKind:
          of: Job
      - equal:
          path: spec.template.spec.containers[0].image
          value: test-hook:2.2.2
