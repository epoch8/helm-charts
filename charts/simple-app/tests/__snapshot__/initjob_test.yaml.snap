All manifests should match snapshot:
  1: |
    apiVersion: batch/v1
    kind: Job
    metadata:
      annotations:
        helm.sh/hook: post-install,post-upgrade
        helm.sh/hook-delete-policy: hook-succeeded,before-hook-creation
        helm.sh/hook-weight: "-1"
      labels:
        app.kubernetes.io/instance: simple
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: simple-app
        app.kubernetes.io/version: 1.16.0
        helm.sh/chart: simple-app-0.0.0
      name: simple-init-job
    spec:
      template:
        spec:
          containers:
            - command:
                - my_service
                - run
              env:
                - name: TEST
                  value: test
              image: test:1.1.1
              imagePullPolicy: IfNotPresent
              name: hook-job
              resources: null
          restartPolicy: Never
          securityContext: {}
          serviceAccountName: simple
